iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -1032,6 +1032,7 @@ private static class IotHubWirePropertyNames
             public const string MqttDiagIdKey = "$.diagid";
             public const string MqttDiagCorrelationContextKey = "$.diagctx";
             public const string InterfaceId = "$.ifid";
+            public const string ComponentName = "$.sub";
         }
```

Comment:
If Message.ComponentName is set for MQTT and not via application property; would it still be set on topic? I don't see that code anywhere.

Reply:
Yes, it would happen here: https://github.com/Azure/azure-iot-sdk-csharp/blob/081a3ecaa49ecc88c88b9d803fcc10dfd5ab5a83/iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs#L1194

We do not need to map each system property explicitly for mqtt, it merges all your application and system properties, and appends them to the topic.

I have also tested this functionality over Mqtt, and the component name is being set as expected.